packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -869,6 +871,13 @@ export interface SdkPackage<TServiceOperation extends SdkServiceOperation> {
   crossLanguagePackageId: string;
   namespaces: SdkNamespace<TServiceOperation>[];
   licenseInfo?: LicenseInfo;
+  /**
+   * The version of the package.
+   * If undefined, the package is not versioned.
+   * If `all`, the package is versioned with all versions.
+   * If a string, the package is versioned with the specified version.
+   */
+  apiVersion?: string;
```

Comment:
I am not sure if `all` is useful here.

I'd assume even if we set `api-version=all` in config, the `api-version` we print out it may still be the `api-version` for the APIs. This depends on the intended use of the output.
- if the output is used for `regen` the SDK, e.g. if it integrated with `tsp-client update`, then `all` is what was expected
- if the output is used for indexing `api-version` -> `sdk version` map, the `all` would be useless